src/ethereum_spec_tools/evm_tools/fixture_loader.py

```diff
@@ -296,6 +296,32 @@ def json_to_tx(self, raw: Any) -> Any:
             hex_to_u256(raw.get("s")),
         ]
 
+        # Cancun and beyond
+        if "maxFeePerBlobGas" in raw:
+            parameters.insert(0, U64(1))
```

Comment:
Inserting by index looks extremely brittle. Could we do something like:

```python
from dataclass import fields

@dataclass
class Foobar:
    nonce: Optional[Uint]
    gasLimit: Optional[Uint]
    # ...

    def to_list(self) -> List[?]:
        parameters = []
        for name in fields(self):
            value = getattr(self, name)
            if value is None:
                continue
            parameters.append(value)
```

Or is that overkill?

Reply:
Personally, I do not think it is brittle in the sense that even a small issue with this will most likely cause every single test to fail and we'll know right away.

That said, I see how this could quickly get out of hand from a maintenance/updating perspective as we add more forks and more parameters. So I would be in favour of doing a bit of refactoring here post cancun.